clients/slack/client.py

```diff
@@ -124,7 +128,8 @@ def _get_channels(self) -> List[dict]:
             has_more = True
             cursor = None
             while has_more:
-                response = self.client.conversations_list(cursor=cursor)
+                response = self.client.conversations_list(cursor=cursor, types='public_channel,private_channel',
```

Comment:
I think it will be better to leave the private_channel out of this.
Private channels should be handled manually by the user because they are sensitive.
If by accident the user gives extra permission to the app, I don't it to be able to join and post on private channels that it wasn't assigned to manually.

Reply:
I'll explain myself better 🙂 
When you create a Slack application, you select the scopes of permissions you grants to it.
We currently don't guide the users to add permissions that allows the application to access private channels, but if by mistake they did grants those permissions (can be done very easily), it will be able to join any channel in the Slack.
So if by mistake the application has over privileges, it will have access to any private channel in the organizatoin (even if the user doesn't have).
This is an edge case, and mainly the user responsibility, but if we can avoid it on our side, I think we should.

The only meaning of this is if a user provides a name of a private channel, it will have to add the application manually to it first.
For public channels, it will join automatically.

Reply:
<img width="656" alt="image" src="https://user-images.githubusercontent.com/30181361/185072078-acfac833-3e86-430f-890e-b1c1a18fe521.png">

Reply:
We should also print the private channels that the app is a part of.
This can be useful for typos, etc.
Say there's a private channel called `data-ops` and the user wrote `dataops`.
If the list of available channels will not show `data-ops`, it will just be confusing to him and he will not assume that `data-ops` would work.

Reply:
Agree!
For documentation sake - looking on Slack application permission scopes confirms that a Slack application can only see a private channels it has been added to.
Approving this solution 🙂